### ISSUE TITLE
Need to choose a logging strategy

### ISSUE THREAD

**Issue by ThrawnCA** [spotbugs/spotbugs#635]
The SpotBugs codebase includes examples of:

- JDK 1.4+ built-in logging, eg in `edu.umd.cs.findbugs.FindBugs` and `edu.umd.cs.findbugs.gui2.MainFrameComponentFactory`
- Multiple incompatible hand-rolled logging setups, based on `edu.umd.cs.findbugs.classfile.IErrorLogger` and `edu.umd.cs.findbugs.log.Logger`
- STDOUT/STDERR logging, all over the place. Note that the SpotBugsBot marks this as a major defect.

We should choose a preferred logging style and stick to it.

**Comment by ThrawnCA** [comment]
Since we don't need complex logging setups given our use case, nor do we need to handle tremendous throughput, and we want maximum compatibility with a variety of codebases, I'm leaning toward JDK standard logging.

**Comment by KengoTODA** [comment]
I personality prefer SLF4J over JUL (JDK standard logging), because JUL does not provide intuitive API. Maybe we can introduce it from v4.0.

**Comment by ThrawnCA** [comment]
I agree that JUL has a very poor API for parameterised logging. Still, it's available without adding any dependencies. If we're OK with adding a dependency, then SLF4J is fine with me. My workplace uses it, it's a good API.

Might be worth starting to replace STDOUT/STDERR with JUL now, then converting to SLF4J later?

**Comment by zolyfarkas** [comment]
+1 for JUL, adding a dependency has its costs, and I don't think is it worth it...

**Comment by hazendaz** [comment]
If this was still not done, didn't look, I would also go with slf4j.  Nearly every OSS uses it that is popular so it only makes sense to follow the heard.  Jul logging sucks and I've never seen a production ready app in the 1000s in my daily life that make such usage.  They are all slf4j.  In fact, lets say we made spotbugs wrapped with spring boot for some fun reason, its going to be slf4j or log4j2.  More so slf4j by default.  A few repos I'm involved with tried years ago to be smarter than the loggers, those all end up causing issues.
